lib/semantics/symbol.cc

```diff
@@ -23,6 +23,28 @@ std::ostream &operator<<(std::ostream &os, const parser::CharBlock &name) {
   return os << name.ToString();
 }
 
+const Scope *ModuleDetails::parent() const {
+  return isSubmodule_ ? &scope_->parent() : nullptr;
+}
+const Scope *ModuleDetails::ancestor() const {
```

Comment:
"ancestor" is a term that could apply to intermediate submodules, too.  If this member function locates the top-level module, a more specific name would be better, I think.

Reply:
This is tricky. The standard refers to "the ancestor module" and "an ancestor module or submodule". So far the code has only dealt with the former, so I've been using "ancestor" for that. But it will be a problem it we do end up needing a name for the latter.